### ISSUE TITLE
LIMIT pushdown when there is group by on distribution column

### ISSUE THREAD

**Issue by saicitus** [citusdata/citus#1589]
For queries with **LIMIT clause** and **group by on distribution column**, LIMIT can be pushed down to each shard. But we don't do this in some cases as below:
 
```
CREATE TABLE test(dist_col integer, other_col integer);
SELECT create_distributed_table('test','dist_col');
```
**In the below query LIMIT is push-downable because there is a group by on partition column, but by default it doesn't get pushed down:**
```
EXPLAIN SELECT count(*) AS count from test group by dist_col order by count LIMIT 1;
                                                               QUERY PLAN
-----------------------------------------------------------------------------------------------------------------------------------------
 Limit  (cost=0.00..0.00 rows=0 width=0)
   ->  Sort  (cost=0.00..0.00 rows=0 width=0)
         Sort Key: COALESCE((pg_catalog.sum((COALESCE((pg_catalog.sum(remote_scan.count))::bigint, '0'::bigint))))::bigint, '0'::bigint)
         ->  HashAggregate  (cost=0.00..0.00 rows=0 width=0)
               Group Key: remote_scan.worker_column_2
               ->  Custom Scan (Citus Real-Time)  (cost=0.00..0.00 rows=0 width=0)
                     Task Count: 32
                     Tasks Shown: One of 32
                     ->  Task
                           Node: host=ec2-34-231-181-49.compute-1.amazonaws.com port=5432 dbname=citus
                           ->  HashAggregate  (cost=43.90..45.90 rows=200 width=12)
                                 Group Key: dist_col
                                 ->  Seq Scan on test_102136 test  (cost=0.00..32.60 rows=2260 width=4)
```
**By adding limit approximation the LIMIT is pushed down:**
```
SET citus.limit_clause_row_fetch_count to 1000;
SET
citus=> EXPLAIN SELECT count(*) AS count from test group by dist_col order by count LIMIT 1;
                                                               QUERY PLAN
-----------------------------------------------------------------------------------------------------------------------------------------
 Limit  (cost=0.00..0.00 rows=0 width=0)
   ->  Sort  (cost=0.00..0.00 rows=0 width=0)
         Sort Key: COALESCE((pg_catalog.sum((COALESCE((pg_catalog.sum(remote_scan.count))::bigint, '0'::bigint))))::bigint, '0'::bigint)
         ->  HashAggregate  (cost=0.00..0.00 rows=0 width=0)
               Group Key: remote_scan.worker_column_2
               ->  Custom Scan (Citus Real-Time)  (cost=0.00..0.00 rows=0 width=0)
                     Task Count: 32
                     Tasks Shown: One of 32
                     ->  Task
                           Node: host=ec2-34-231-181-49.compute-1.amazonaws.com port=5432 dbname=citus
                           **->  Limit  (cost=53.54..54.04 rows=200 width=12)**
                                 ->  Sort  (cost=53.54..54.04 rows=200 width=12)
                                       Sort Key: (count(*))
                                       ->  HashAggregate  (cost=43.90..45.90 rows=200 width=12)
                                             Group Key: dist_col
                                             ->  Seq Scan on test_102136 test  (cost=0.00..32.60 rows=2260 width=4)
```
The end-user might think that it is an approximation, but actually it is not an approximation, because the group by on dist_col pushes the complete aggregation to the worker shards.
Same with the below query(no order by)

SELECT count(*) AS count from test group by dist_col LIMIT 1;

For this query setting the limit approximation doesn't push the limit down.
The only way to push the limit down the query is to add a random ORDER BY on non-aggregate column.

SELECT count(*) AS count from test group by dist_col ORDER BY dist_col LIMIT 1;

**Comment by saicitus** [comment]
Ran into this in 2 prospect engagements.

**Comment by marcocitus** [comment]
Duplicate of #606 . @aamederen may have more context.

**Comment by pykello** [comment]
Fixed by #1641.
